docs(calendar,card,coachmark,colorarea): site docs to storybook

mdt2 opened the pull request:
## Description

This migrates the documentation from the deprecated docs site over to Storybook, including showing the variants on the Storybook Docs page.

- Calendar - got a new MDX page and one MDX-only story
- Card - got a new MDX page and some MDX-only stories
- Coachmark - got a new MDX page and some extra parameters to make sure "with media" variant shows whole component on Docs page
- Colorarea - got a new MDX page and one MDX-only story

This PR doesn't need a changeset since it's docs-only.

## How and where has this been tested?

Please tag yourself on the tests you've marked complete to confirm the tests have been run by someone other than the author.

### Validation steps

- [x] The [Calendar storybook](https://pr-2783--spectrum-css.netlify.app/preview/?path=/docs/components-calendar--docs) docs include all necessary info when compared to the [docs site](https://pr-2783--spectrum-css.netlify.app/calendar)
- [x] The [Card storybook](https://pr-2783--spectrum-css.netlify.app/preview/?path=/docs/components-card--docs) docs include all necessary info when compared to the [docs site](https://pr-2783--spectrum-css.netlify.app/card)
  - [x] Also check that the variants from the [docs site asset preview](https://pr-2783--spectrum-css.netlify.app/card-asset) and [gallery pages](https://pr-2783--spectrum-css.netlify.app/card-gallery) are covered.
- [ ] The [Coachmark storybook](https://pr-2783--spectrum-css.netlify.app/preview/?path=/docs/components-coach-mark--docs) docs include all necessary info when compared to the [docs site](https://pr-2783--spectrum-css.netlify.app/coachmark)
- [ ] The [Colorarea storybook](https://pr-2783--spectrum-css.netlify.app/preview/?path=/docs/components-color-area--docs) docs include all necessary info when compared to the [docs site](https://pr-2783--spectrum-css.netlify.app/colorarea)

### Regression testing

Validate:

1. The documentation pages for at least two other components are still loading, including:

- [ ] The pages render correctly, are accessible, and are responsive.

2. If components have been modified, VRTs have been run on this branch:

- [ ] VRTs have been run and looked at.
- [ ] Any VRT changes have been accepted (by reviewer and/or PR author), or there are no changes.


## To-do list

- [x] I have read the [contribution guidelines](/.github/CONTRIBUTING.md).
- [x] I have updated relevant storybook stories and templates.
- [x] If my change impacts **documentation**, I have updated the documentation accordingly.
- [ ] ✨ This pull request is ready to merge. ✨

rise-erpelding commented on components/card/stories/card.mdx:
```diff
@@ -0,0 +1,62 @@
+
+<Canvas of={CardStories.Quiet} />
+
+### File
```
This and the several variants below it could use "Quiet" in the heading so as not to be confused with their non-quiet counterparts.

rise-erpelding commented on components/card/stories/card.mdx:
```diff
@@ -0,0 +1,62 @@
+## Gallery
+A gallery card for an image.
+
+<Canvas of={CardStories.Gallery} />
```
Is there a reason this one looks different from the Docs site with the placement of the "jpg" and missing date? (Doesn't seem like a problem for this PR though.)
![image](https://github.com/adobe/spectrum-css/assets/54716846/18ef9012-7756-41ab-8f7f-5fd961524f6e)

mdt2 replied:
Great question. It looks like this is because the "Subtitle" option was deprecated ([migration guide note](https://pr-2783--spectrum-css.netlify.app/card#migrationguide)), and that's what the gallery example was showing before.

rise-erpelding commented on components/coachmark/stories/coachmark.mdx:
```diff
@@ -0,0 +1,20 @@
+
+## Standard
+
+<Canvas of={CoachmarkStories.Default} />
```
Also not necessarily a problem for this PR since it existed before but are you also seeing the top edge and left drop shadow being cut off like this? 🙃 
![image](https://github.com/adobe/spectrum-css/assets/54716846/9243de1d-e820-4fbb-b0f1-09248eebe648)

mdt2 replied:
It looks like this is [happening on prod](https://opensource.adobe.com/spectrum-css/preview/?path=/docs/components-coach-mark--docs) as well 🤔 But only on the docs page. It seems like the component butts right up to the container, so I went ahead and added some small top margin to the component in the template to fix this.

rise-erpelding reviewed (commented):
Hello (again)! Thanks for all the progress you're making with these, I'm excited to see them coming through! There's one thing that I noted that could be clarified a bit better as well as some weird inconsistencies that are probably out of scope for this card that I noted in the comments below.

The only other additional question I have for this one is that I noted that **Color area** includes the disabled variant and **Card** includes a focused variant and the **Calendar** docs include both disabled and focused variants but on Storybook neither variant is included, was there a reason for leaving those out of the .mdx docs for **Calendar**?

mdt2 commented:
Good callouts @rise-erpelding! I didn't originally add disabled or focused for Calendar because those felt less like variants and more like states, and in my head I was like "states should maybe just be shown with controls." But I clearly didn't think that with much conviction since I added them to the other two 😅 So I went ahead and added disabled and focused to Calendar too 👍

mdt2 commented on components/calendar/stories/template.js:
```diff
@@ -193,6 +194,9 @@ export const Template = ({
 					);
+					const isFocused = thisDay === 5;
+
+					console.log(thisDay, isFocused);
```
```suggestion
```

rise-erpelding reviewed (approved):
Looking great! Thanks, Melissa!
